brayns/io/loaders/volume/VolumeLoader.cpp

```diff
@@ -19,182 +19,222 @@
 
 #include "VolumeLoader.h"
 
-#include <brayns/engine/Model.h>
-#include <brayns/engine/Scene.h>
-#include <brayns/engine/Volume.h>
-#include <brayns/utils/StringUtils.h>
+#include <brayns/engine/components/Volumes.h>
+#include <brayns/engine/systems/GenericBoundsSystem.h>
+#include <brayns/engine/systems/VolumeCommitSystem.h>
+#include <brayns/engine/systems/VolumeInitSystem.h>
+#include <brayns/engine/volume/types/RegularVolume.h>
+
+#include <brayns/utils/FileReader.h>
+#include <brayns/utils/parsing/Parser.h>
+#include <brayns/utils/string/StringExtractor.h>
+#include <brayns/utils/string/StringSplitter.h>
+#include <brayns/utils/string/StringTrimmer.h>
 
 #include <filesystem>
-#include <fstream>
-#include <map>
-#include <sstream>
-#include <string>
+#include <unordered_map>
 
-namespace brayns
-{
 namespace
 {
-template<size_t M, typename T>
-std::string to_string(const glm::vec<M, T> &vec)
+class MhdParser
 {
-    std::stringstream stream;
-    stream << vec;
-    return stream.str();
-}
+public:
+    static std::unordered_map<std::string, std::string> parse(const std::string &filename)
+    {
+        auto content = brayns::FileReader::read(filename);
+        auto view = std::string_view(content);
+
+        auto result = std::unordered_map<std::string, std::string>();
+
+        auto ctr = 1ul;
+        while (!view.empty())
```

Comment:
I had written a FileStream class to combine iteration with line number just in case.

Reply:
Addressed. 
There is an unhandled corner case on the straem into which I ran (empty line at the end of the file). I think it would be nice to handle it since its pretty common.